src/app/favourite/favourite.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+import { FirebaseService } from '../firebase.service';
+
+@Component({
+  selector: 'app-favourite',
+  templateUrl: './favourite.component.html',
+  styleUrls: ['./favourite.component.scss']
+})
+export class FavouriteComponent implements OnInit {
+  recipe: any;
+  isLiked = false;
+
+  constructor(private fbService: FirebaseService) { }
+
+  ngOnInit(): void {
+    this.fbService.getRecipe(0).valueChanges().subscribe((recipe: any) => {
+      this.recipe = recipe;
+      if (this.recipe.saved) {
+        this.isLiked = true;
+      }
+    });
+  }
+
+  addToFavorite(): void {
+    this.fbService.updateProps(0, { saved: !this.recipe.saved }).then(() => {
+      this.recipe.saved = this.recipe.saved ? !this.recipe.saved : this.recipe.saved;
+      this.isLiked = !this.isLiked;
```

Comment:
again, you don't need this variable since you already have this information saved